openff/evaluator/protocols/forcefield.py

```diff
@@ -1052,3 +1063,54 @@ def _execute(self, directory, available_resources):
             )
 
         super(BuildTLeapSystem, self)._execute(directory, available_resources)
+
+
+@workflow_protocol()
+class BuildFoyerSystem(TemplateBuildSystem):
+    """Parameterize a set of molecules with a Foyer force field source"""
+
+    def _parameterize_molecule(self, molecule, force_field_source, cutoff):
+        """Parameterize the specified molecule.
+
+        Parameters
+        ----------
+        molecule: openff.toolkit.topology.Molecule
+            The molecule to parameterize.
+        force_field_source: FoyerForceFieldSource
+            The foyer source which describes which parameters to apply.
+
+        Returns
+        -------
+        openmm.System
+            The parameterized system.
+        """
+        import mdtraj as md
+        from foyer import Forcefield
+        from openff.interchange import Interchange
+        from openff.toolkit import Topology
+
+        topology: Topology = molecule.to_topology()
+        topology.mdtop = md.Topology.from_openmm(topology.to_openmm())
+
+        force_field: Forcefield
+        if force_field_source.foyer_source.lower() == "oplsaa":
+            force_field = Forcefield(name="oplsaa")
+        else:
+            force_field = Forcefield(forcefield_files=force_field_source.foyer_source)
+
+        interchange = Interchange.from_foyer(topology=topology, force_field=force_field)
+        interchange["vdW"].mixing_rule = "lorentz-berthelot"
```

Comment:
I don't think we should be doing this - AFAIR OPLS-AA is defined to use geometric mixing rules and we should try to respect that. I've put some work recently into making sure Interchange can handle it. Have you run into issues using it here? Needing to split `NonbondedForce` out into a bunch of custom forces is what would cause headaches, if I had to guess.

Reply:
Good point, I got this code from https://docs.openforcefield.org/projects/interchange/en/stable/using/migrating.html where LB mixing were forced onto OPLS-AA

Reply:
Once again my error, and once again thanks for bringing this up!